dibi/libs/DibiFluent.php

```diff
@@ -315,7 +315,7 @@ public function execute($return = NULL)
 	 */
 	public function fetch()
 	{
-		if ($this->command === 'SELECT') {
+		if ($this->command === 'SELECT' && $this->_export('limit') === NULL) {
```

Comment:
Seems like I have a solution for this and I bet you're gonna like it...

I'll try to add it here or create another PR otherwise